ffi/diplomat/src/datetime.rs

```diff
@@ -5,244 +5,385 @@
-    // from TimeFormatter.
-    pub struct ICU4XTimeFormatter(pub TimeFormatter);
-
-    pub enum ICU4XTimeLength {
-        Full,
-        Long,
-        Medium,
-        Short,
-    }
+    /// An ICU4X DateTime object capable of containing a ISO-8601 date and time.
+    #[diplomat::rust_link(icu::calendar::DateTime, Struct)]
+    pub struct ICU4XIsoDateTime(pub DateTime<Iso>);
 
-    impl ICU4XTimeFormatter {
-        /// Creates a new [`ICU4XTimeFormatter`] from locale data.
-        #[diplomat::rust_link(icu::decimal::TypedDateFormatter::try_new_unstable, FnInStruct)]
+    impl ICU4XIsoDateTime {
+        /// Creates a new [`ICU4XIsoDateTime`] from the specified date and time.
+        #[diplomat::rust_link(icu::DateTime::new_iso_datetime, FnInStruct)]
         pub fn try_new(
-            provider: &ICU4XDataProvider,
-            locale: &ICU4XLocale,
-            length: ICU4XTimeLength,
-        ) -> DiplomatResult<Box<ICU4XTimeFormatter>, ICU4XError> {
-            use icu_provider::serde::AsDeserializingBufferProvider;
-            let provider = provider.0.as_deserializing();
-
-            let locale = locale.to_datalocale();
-            let length = match length {
-                ICU4XTimeLength::Full => length::Time::Full,
-                ICU4XTimeLength::Long => length::Time::Long,
-                ICU4XTimeLength::Medium => length::Time::Medium,
-                ICU4XTimeLength::Short => length::Time::Short,
-            };
+            year: i32,
+            month: u8,
+            day: u8,
+            hour: u8,
+            minute: u8,
+            second: u8,
+            nanosecond: u32,
+        ) -> DiplomatResult<Box<ICU4XIsoDateTime>, ICU4XError> {
+            let nanosecond = try_icu4x!(nanosecond.try_into());
+            DateTime::new_iso_datetime(year, month, day, hour, minute, second)
+                .map(|mut dt| {
+                    dt.time.nanosecond = nanosecond;
+                    Box::new(ICU4XIsoDateTime(dt))
+                })
+                .map_err(Into::into)
+                .into()
+        }
 
-            TimeFormatter::try_new_unstable(&provider, &locale, length)
-                .map(|tf| Box::new(ICU4XTimeFormatter(tf)))
+        /// Creates a new [`ICU4XIsoDateTime`] from an [`ICU4XIsoDate`] and [`ICU4XTime`] object
+        #[diplomat::rust_link(icu::calendar::DateTime::new, FnInStruct)]
+        pub fn new_from_date_and_time(
+            date: &ICU4XIsoDate,
+            time: &ICU4XTime,
+        ) -> Box<ICU4XIsoDateTime> {
+            let dt = DateTime::new(date.0, time.0);
+            Box::new(ICU4XIsoDateTime(dt))
+        }
+
+        /// Construct from the minutes since the local unix epoch for this date (Jan 1 1970, 00:00)
+        #[diplomat::rust_link(
+            icu::calendar::DateTime::from_minutes_since_local_unix_epoch,
+            FnInStruct
+        )]
+        pub fn from_minutes_since_local_unix_epoch(
```

Comment:
Observation: another case where we are inconsistent with `new`, `try_new`, `create`, and just `from`